src/index.js

```diff
@@ -1,70 +1,699 @@
-// even if this in another domain here, but it works as an example
-async function authenticate(login, password) {
-  // setup cookies
-  log('info', 'Setup cookies...')
-  await lib.setupCookies()
+const baseUrl = 'https://www.macif.fr'
+// let FORCE_FETCH_ALL = false
 
-  log('info', 'Logging in...')
-  await lib.login(login, password)
-}
+const personnalInfos = []
+const personIdentity = []
+const attestationsInfos = []
+const schedulesInfos = []
 
-async function fetchPaymentNotices() {
-  const res = JSON.parse(await lib.fetchPaymentNotices())
-  return lib.parsePaymentNotices(res)
+var openProxied = window.XMLHttpRequest.prototype.open
+window.XMLHttpRequest.prototype.open = function () {
+  var originalResponse = this
+  if (arguments[1].includes('/espace-client/contacts/macif')) {
+    originalResponse.addEventListener('readystatechange', function () {
+      if (originalResponse.readyState === 4) {
+        const jsonInfos = JSON.parse(originalResponse.responseText)
+        personnalInfos.push(jsonInfos)
+      }
+    })
+    return openProxied.apply(this, [].slice.call(arguments))
+  }
+  if (arguments[1].match(/\/internet-personne-rest\/personnes\/\d+/g)) {
+    originalResponse.addEventListener('readystatechange', function () {
+      if (originalResponse.readyState === 4) {
+        const jsonInfos = JSON.parse(originalResponse.responseText)
+        personIdentity.push(jsonInfos)
+      }
+    })
+    return openProxied.apply(this, [].slice.call(arguments))
+  }
+  return openProxied.apply(this, [].slice.call(arguments))
 }
 
-async function fetchInsuranceCertificates() {
-  const res = JSON.parse(await lib.fetchInsuranceCertificates())
-  return lib.parseInsuranceCertificates(res)
+const fetchOriginal = window.fetch
+window.fetch = async (...args) => {
+  const response = await fetchOriginal(...args)
+  if (
+    args[0].url &&
+    args[0].url === 'https://ssm.macif.fr/internet-contrat-rest/v2/attestations'
+  ) {
+    await response
+      .clone()
+      .json()
+      .then(body => {
+        attestationsInfos.push(body)
+        return response
+      })
+      .catch(err => {
+        // eslint-disable-next-line no-console
+        console.log(err)
+        return response
+      })
+  }
+  if (
+    args[0].url &&
+    args[0].url ===
+      'https://ssm.macif.fr/internet-espaceclient-rest/personnes/1/document/avisecheance'
+  ) {
+    await response
+      .clone()
+      .json()
+      .then(body => {
+        schedulesInfos.push(body)
+        return response
+      })
+      .catch(err => {
+        // eslint-disable-next-line no-console
+        console.log(err)
+        return response
+      })
+  }
+  return response
 }
 
-const parseDate = x => {
-  return parse(x, 'YYYY-MM-DD')
-}
+class MacifContentScript extends ContentScript {
+  onWorkerReady() {
+    window.addEventListener('DOMContentLoaded', () => {
```

Comment:
Are we sure, DOMContentLoaded is always triggered after onWorkerReady ?

Reply:
Yes, im' sure this is needed for this konnector, at least until now I've manage to intercept the credentials only when using this listener 👍

Reply:
What I wanted to say is that this event could also be triggered before onWorkerReady and you would miss it.

Reply:
Ok I see, changes incoming 👍